src/components/WebcamCapture/WebcamCapture.js

```diff
@@ -5,21 +5,56 @@ import PropTypes from 'prop-types';
 import './WebcamCapture.css';
 
 class WebcamCapture extends Component {
+  constructor() {
+    super();
+    global.injectWebcam = ({isDetecting, cameraConnected}) => {
+      this.setState({
+        fakeWebcam: true,
+        isDetecting,
+        cameraConnected
+      });
+      if (this.props.onConnect) this.props.onConnect();
+    };
+  }
+
   state = {
+    fakeWebcam: false,
     isDetecting: true,
     cameraConnected: false
   };
 
   webcam = React.createRef();
 
-  requestScreenshot = () => {
-    if (!this.webcam.current) return Promise.reject('Failed to load webcam.');
-    const screenshot = this.webcam.current.getScreenshot();
-    if (screenshot === null) return Promise.reject('Failed to load webcam.');
+  requestFakeScreenshot = () =>
+    new Promise((resolve, reject) => {
+      const input = this.fileInput;
+      if (input.files == null || input.files[0] == null) {
+        return reject(new Error('No files present'));
+      }
+      const reader = new FileReader();
+      reader.onload = ({target: {result}}) => {
+        resolve(result);
+      };
```

Comment:
Could this be simplified?

```
reader.onload = {target: {result}} => resolve(result);
```

Reply:
The above throws an error.
Another way to write it could be
```javascript
reader.onload = (event) => {
    resolve(event.target.result);
};
```